src/map.c

```diff
@@ -170,7 +172,11 @@ SEXP pmap_impl(SEXP env, SEXP l_name_, SEXP f_name_, SEXP type_, SEXP progress)
     if (n == -1) {
       n = nj;
     } else if (nj != n) {
-      stop_bad_element_length(j_val, j + 1, n, NULL, ".l", true);
+      r_abort_call(
+        error_call,
+        "`.l[[%i]]` must have length 1 or %i, not %i.",
```

Comment:
```suggestion
        "`.l[[%i]]` must be length 1 or %i, not %i.",
```

Reply:
I'm not going to do this here in favour of using vctrs errors in #957